### PR TITLE
Feature/game logic

### PR DESCRIPTION
**Pull request by jpdf00**
Create game_logic.rb.
Integrate game_logic with the user interface.
 - Game accepts 2 players.
 - Game stops at a win and prints the winning player.
 - Game runs for 9 turns and ends in a draw if no player won.
 - Used Magic Square with prime numbers to solve who is the winner.

### CODE REVIEW

File: lib/game_logic.rb [hunk 1]
```diff
@@ -0,0 +1,107 @@
...
+  end
+
+  def execute_turn(selected_index)
+    outter = nil
+    inner = nil
+    @board.printable_board.each_with_index do |item, outter_index|
+      inner = inner.nil? ? item.find_index(selected_index) : break
+      outter = outter_index
+    end
+    raise 'The selected index is not available' if inner.nil?
```
Comment (rindrajosia) [inline lib/game_logic.rb:94]: This line raises an error as a result, the game stop.

File: lib/game_logic.rb [hunk 2]
```diff
@@ -0,0 +1,107 @@
...
+    @player_id = player_id
+  end
+
+  def update_player_board(index1, index2, cell_number)
+    @array[index1][index2] = cell_number * @player_id
+  end
+
+  def winner?(magic_number)
+    player_magic = magic_number * @player_id
+    # Test the sum of all the rows
```
Comment (rindrajosia) [inline lib/game_logic.rb]: No inline comment

**Review by rindrajosia — CHANGES_REQUESTED** [review]
CHANGES REQUIRED
HELLO!! You did a great job in this millstone very nice work  but there are some changes that you have to make, please check below:

REQUIRED:
- [x] The application should not exit/stop if the player chooses an occupied position.
- [x] No inline comments.

After addressing those minor bugs submit another CRR.

Happy Coding!
Rindra Josia

**Review by LamiaSristy — APPROVED** [review]
# Status: Approved. 💯 

Hello Team @jpdf00  and, @NoTengoBattery 

Congratulations, your project is approved 👏 👏 👏

You may merge, good luck with your next milestone. 🚀

Feel free to reach me at @lamiasristy 🙋

**Pull request merged**